page.tesk.Refine.json

```diff
@@ -0,0 +1,163 @@
+                }
+            ]
+        },
+        {
+            "name": "dconf",
+            "buildsystem": "meson",
+            "config-opts": [
+                "-Dbash_completion=false",
+                "-Dman=false"
+            ],
+            "cleanup": [
+                "/include",
+                "/lib/pkgconfig",
+                "/libexec",
+                "/share/dbus-1"
+            ],
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://download.gnome.org/sources/dconf/0.40/dconf-0.40.0.tar.xz",
+                    "sha256": "cf7f22a4c9200421d8d3325c5c1b8b93a36843650c9f95d6451e20f0bcb24533"
+                },
+                {
+                    "type": "patch",
+                    "path": "dconf-override.patch"
+                }
+            ]
+        },
+        {
+            "name": "libsass",
+            "buildsystem": "meson",
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/lazka/libsass.git",
+                    "commit": "aac79dccd3c8f7e8f22125f87a119f3b1ee9d487",
+                    "branch": "meson"
+                }
+            ]
+        },
+        {
+            "name": "sassc",
+            "buildsystem": "meson",
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/lazka/sassc.git",
+                    "commit": "a1950c2d95ea4c051feb90bb1f43559fbb54bf36",
+                    "branch": "meson"
+                }
+            ]
+        },
+        {
+            "name": "libadwaita",
+            "buildsystem": "meson",
+            "config-opts": [
+                "-Dexamples=false",
+                "-Dtests=false"
+            ],
+            "cleanup": [
+                "/include",
+                "/lib/pkgconfig",
+                "/libexec",
+                "/share/dbus-1"
+            ],
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://gitlab.gnome.org/GNOME/libadwaita.git",
+                    "commit": "2ea59b267b2e10d18050f395d861a58b96ad9b5c"
+                }
+            ]
+        },
+        {
+            "name": "refine",
+            "builddir": true,
+            "buildsystem": "meson",
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://gitlab.gnome.org/TheEvilSkeleton/Refine.git",
+                    "tag": "0.1.0",
+                    "commit": "8b8595bcf2c7c40ac36754bca3563d6e76235299"
+                }
+            ],
+            "post-install": [
+                "sed -i 's,^Exec=refine$,Exec=start-refine,g' /app/share/applications/$FLATPAK_ID.desktop"
+            ]
+        },
+        {
+            "name": "scripts",
```

Comment:
this should be part of the main module.

Reply:
How would I go about doing that?

Reply:
Merge the sources, add the command to post-install

Reply:
The manifest is currently the following:
```json
        {
            "name": "refine",
            "builddir": true,
            "buildsystem": "meson",
            "post-install": [
                "ls -la",
                "install -Dm 755 start-refine.sh /app/bin/start-refine",
                "desktop-file-edit --set-key='Exec' --value='/app/bin/start-refine' /app/share/applications/$FLATPAK_ID.desktop"
            ],
            "sources": [
                {
                    "type": "git",
                    "url": "https://gitlab.gnome.org/TheEvilSkeleton/Refine.git",
                    "tag": "0.1.0",
                    "commit": "8b8595bcf2c7c40ac36754bca3563d6e76235299"
                },
                {
                    "type": "file",
                    "path": "start-refine.sh"
                }
            ]
        }
```

but I'm getting the error "cannot stat 'start-refine.sh': No such file or directory". Running `ls -la` in post-install gives me the following result:
```
total 36
drwxr-xr-x 1 theevilskeleton theevilskeleton  234 Dec 31 09:52 .
drwxr-xr-x 1 theevilskeleton theevilskeleton  312 Dec 31 09:52 ..
-rw-r--r-- 1 theevilskeleton theevilskeleton 8160 Dec 31 09:52 build.ninja
-rw-r--r-- 1 theevilskeleton theevilskeleton 5355 Dec 31 09:52 compile_commands.json
drwxr-xr-x 1 theevilskeleton theevilskeleton  346 Dec 31 09:52 data
-rw-r--r-- 1 theevilskeleton theevilskeleton   92 Dec 31 09:52 .gitignore
-rw-r--r-- 1 theevilskeleton theevilskeleton  108 Dec 31 09:52 .hgignore
drwxr-xr-x 1 theevilskeleton theevilskeleton  496 Dec 31 09:52 meson-info
drwxr-xr-x 1 theevilskeleton theevilskeleton   56 Dec 31 09:52 meson-logs
drwxr-xr-x 1 theevilskeleton theevilskeleton  226 Dec 31 09:52 meson-private
-rw-r--r-- 1 theevilskeleton theevilskeleton 1136 Dec 31 09:52 .ninja_deps
-rw-r--r-- 1 theevilskeleton theevilskeleton  483 Dec 31 09:52 .ninja_log
drwxr-xr-x 1 theevilskeleton theevilskeleton    0 Dec 31 09:52 po
drwxr-xr-x 1 theevilskeleton theevilskeleton  118 Dec 31 09:52 refine
```

Did I miss something?

Reply:
`install -Dm 755 ../start-refine.sh`

value should be just the name not the full path.